exercises/robot-name/robot-name-test.rkt

```diff
@@ -0,0 +1,38 @@
+#lang racket
+(require racket/class
+         "robot-name.rkt")
+
+(module+ test
+  (require rackunit
+           rackunit/text-ui)
+
+  (define proper-robot-name #px"\\p{Lu}{2}\\p{Nd}{4}")
+  
+  (define (check-proper-robot-name r)
+    (check-regexp-match proper-robot-name (send r get-name)))
+  
+  (define robot-name-suite 
+    (test-suite
+     "Check proper operation of 'robots'"
+     
+     (test-case
+      "Check that robot names conforms to the expected standard."
+      (check-proper-robot-name (new robot%)))
+
+     (test-case
+      "Check that resetting a robot name also gives a proper name."
+      (let ([robby (new robot%)])
+        (check-regexp-match proper-robot-name (send robby get-name))
+        (send robby reset)
+        (check-regexp-match proper-robot-name (send robby get-name))))
```

Comment:
This test should probably also check that it is not the same name after resetting

Reply:
The Python version also reset the random seed ... I was wondering if that was also worth doing?

Reply:
Hmm, I don't think it should be necessary.

When I did this exercise in the scheme track (it is now deprecated until we rework it) I wrote test extensions, that reset 1 robot 500 times (I made resetting free the name in the pool) before generating all possible names minus 1 for the bot still in existence.

It didn't take a very long time, so as long as racket has at least as good of a random implementation as Guile it shouldn't be necessary to reset the seed.  My method of generating a name was basically identical to your semantically.

Reply:
This is off topic but the Scheme track seems to not be actively mentored?
I'm not crying myself to sleep over it - but I think I submitted a solution to "leap" more than three months ago.

I'd like to branch out a bit from Racket, mostly because broader exposure tends to help me appreciate the finer details. Guile seemed like a natural direction to go in. I'm not great at Guile, but later on I'd like to see about helping you guys make the Scheme track awesome.

Reply:
Thanks for your enthusiasm.  There are two of us mentoring the Scheme track surrently, but we just started like a wee ago (maybe 2).

I don't think I saw your solution anywhere.  If you send me the solution id I will check it out :smile:

Reply:
I think the missing exercise is my fault ... my Ubuntu box died and I think I am misremembering things. I'll probably resurrect it and start again soon.

Reply:
We would love to see your submissions on the scheme track!